agent/test/unit/com/thoughtworks/go/agent/AgentWebSocketClientControllerTest.java

```diff
@@ -159,19 +156,95 @@ public void processSetCookieAction() throws IOException, InterruptedException {
 
     @Test
     public void processAssignWorkAction() throws IOException, InterruptedException {
+        SystemEnvironment env = new SystemEnvironment();
         ArgumentCaptor<Message> argumentCaptor = ArgumentCaptor.forClass(Message.class);
         agentController = createAgentController();
         agentController.init();
         agentController.process(new Message(Action.assignWork, MessageEncoding.encodeWork(new SleepWork("work1", 0))));
         assertThat(agentController.getAgentRuntimeInfo().getRuntimeStatus(), is(AgentRuntimeStatus.Idle));
 
-        verify(webSocketSessionHandler).sendAndWaitForAcknowledgement(argumentCaptor.capture());
+        verify(webSocketSessionHandler, times(1)).sendAndWaitForAcknowledgement(argumentCaptor.capture());
         verify(artifactsManipulator).setProperty(null, new Property("work1_result", "done"));
 
-        Message message = argumentCaptor.getValue();
+        Message message = argumentCaptor.getAllValues().get(0);
+        assertThat(message.getAcknowledgementId(), notNullValue());
+        assertThat(message.getAction(), is(Action.ping));
+        assertThat(message.getData(), is(MessageEncoding.encodeData(agentController.getAgentRuntimeInfo())));
+
+    }
+
+    @Test
+    public void processAssignWorkActionWitConsoleLogsThroughhWebsockets() throws IOException, InterruptedException {
+        SystemEnvironment env = new SystemEnvironment();
+        env.set(SystemEnvironment.WEBSOCKET_ENABLED, true);
+        env.set(SystemEnvironment.CONSOLE_LOGS_THROUGH_WEBSOCKET_ENABLED, true);
+        ArgumentCaptor<Message> argumentCaptor = ArgumentCaptor.forClass(Message.class);
+        agentController = createAgentController();
+        agentController.init();
+        agentController.process(new Message(Action.assignWork, MessageEncoding.encodeWork(new SleepWork("work1", 0))));
+        assertThat(agentController.getAgentRuntimeInfo().getRuntimeStatus(), is(AgentRuntimeStatus.Idle));
+
+        verify(webSocketSessionHandler, times(2)).sendAndWaitForAcknowledgement(argumentCaptor.capture());
+        verify(artifactsManipulator).setProperty(null, new Property("work1_result", "done"));
+
+        Message message = argumentCaptor.getAllValues().get(1);
         assertThat(message.getAcknowledgementId(), notNullValue());
         assertThat(message.getAction(), is(Action.ping));
         assertThat(message.getData(), is(MessageEncoding.encodeData(agentController.getAgentRuntimeInfo())));
+
+        Message message2 = argumentCaptor.getAllValues().get(0);
+        assertThat(message2.getAcknowledgementId(), notNullValue());
+        assertThat(message2.getAction(), is(Action.consoleOut));
+        assertThat(message2.getData(), is(MessageEncoding.encodeData(new ConsoleTransmission("Sleeping for 0 milliseconds", new JobIdentifier()))));
+        env.set(SystemEnvironment.WEBSOCKET_ENABLED, false);
```

Comment:
If one of the assertions fail, resetting of the properties will not happen, causing some other tests to fail. Mocked `systemEnvironment` is already being passed along to the controller, may be setup the expectations according to the test, and use the same instead of instantiating a new one in [AgentWebSocketClientController#runBuild](https://github.com/kierarad/gocd/blob/2a909964c401ae088def31da815b56e7d6caf475/agent/src/com/thoughtworks/go/agent/AgentWebSocketClientController.java#L163)